cluster/addons/addon-manager/kube-addons.sh

```diff
@@ -217,8 +217,9 @@ start_addon /opt/namespace.yaml 100 10 "" &
 token_found=""
 while [ -z "${token_found}" ]; do
   sleep .5
-  token_found=$(${KUBECTL} ${KUBECTL_OPTS} get --namespace="${SYSTEM_NAMESPACE}" serviceaccount default -o go-template="{{with index .secrets 0}}{{.name}}{{end}}")
-  if [[ $? -ne 0 ]]; then
+  token_found=$(${KUBECTL} "${KUBECTL_OPTS}" get --namespace="${SYSTEM_NAMESPACE}" serviceaccount default -o go-template="{{with index .secrets 0}}{{.name}}{{end}}")
+if ! $token_found;
```

Comment:
This is not equivalent behavior, you're now checking the value of the above command's stdout instead of its exit code

What shellcheck failure are you addressing here?